frontend/src/lib/components/VisibilitySensor/VisibilitySensor.tsx

```diff
@@ -0,0 +1,23 @@
+import { useActions } from 'kea'
+import React, { useEffect, useRef } from 'react'
+import { visibilitySensorLogic } from './visibilitySensorLogic'
+
+interface VisibilityProps {
+    id: string
+    offset: number
+    children: React.ReactNode | null
+}
+
+export function VisibilitySensor({ id, offset, children }: VisibilityProps): JSX.Element {
+    const { scrolling } = useActions(visibilitySensorLogic({ id, offset }))
+
+    const ref = useRef<HTMLDivElement | null>(null)
+
+    useEffect(() => {
+        const element = ref.current
+        document.addEventListener('scroll', () => scrolling(element))
+        return () => document.removeEventListener('scroll', () => scrolling(element))
+    })
```

Comment:
Missing a dep list 🤞 the eslint annotations hightlight this as well

Reply:
sorry, didn't get this?

Reply:
Maybe not needed actually, just commenting that there isn't a dependency array specified here. But possibly should depend on `ref.current` ?

Reply:
Indeed, otherwise this will run in every render, I'll fix this

Reply:
Interesting, I would've thought this would run only once if it has no dep list?

Reply:
Ah, I need an empty dep list for that 🤦 .